app/handlers/retrieveResults.py

```diff
@@ -16,5 +21,32 @@ class RetrieveResultsHandler(tornado.web.RequestHandler):
 
     @apiError error_message The error message to display.
     """
+
+    def post(self):
+        self.identifier = self.get_body_arguments("identifier")
+
     def get(self):
-        self.finish("Retrieve Results")
\ No newline at end of file
+        identifier = self.get_body_arguments("identifier")
+        project_path = get_project_path(identifier[0])
+        file_path = os.path.join(project_path, 'final_videos')
+        file_name = os.path.join(project_path, 'results.zip')
+
+        zipf = zipfile.ZipFile(file_name, 'w', zipfile.ZIP_DEFLATED)
+        for root, dirs, files in os.walk(file_path):
+            for file in files:
+                zipf.write(os.path.join(file_path, file))
+        zipf.close()
+
+        self.set_header('Content-Type', 'application/octet-stream')
+        self.set_header('Content-Description', 'File Transfer')
+        self.set_header('Content-Disposition', 'attachment; filename=' + file_name)
+        with open(file_name, 'rb') as f:
+            try:
+                while True:
+                    data = f.read(2048)
+                    if not data:
+                        break
+                    self.write(data)
+                self.finish()
```

Comment:
Is there a way to do some form of checksum testing on the individual files to know if the content has changed. If so, then re-zip and send those changed files. Otherwise, send nothing.